src/components/App.js

```diff
@@ -90,25 +89,62 @@ export default class App extends React.Component {
       songHistory: []
     };
 
-    // Keyboard shortcuts
-    this.keyMap = {
-      TOGGLE_PLAY: ['space', 'k'],
-      INCREASE_VOLUME: 'up',
-      DECREASE_VOLUME: 'down'
-    };
-
-    // Keyboard shortcut handlers
-    this.handlers = {
-      TOGGLE_PLAY: () => this.togglePlay(),
-      INCREASE_VOLUME: () => this.increaseVolume(),
-      DECREASE_VOLUME: () => this.decreaseVolume()
-    };
-
     this.togglePlay = this.togglePlay.bind(this);
     this.setUrl = this.setUrl.bind(this);
     this.setTargetVolume = this.setTargetVolume.bind(this);
     this.getNowPlaying = this.getNowPlaying.bind(this);
     this.updateVolume = this.updateVolume.bind(this);
+    this.increaseVolume = this.increaseVolume.bind(this);
+    this.decreaseVolume = this.decreaseVolume.bind(this);
+
+    // Keyboard handlers
+    this.addKeyboardHotKeysListener = this.addKeyboardHotKeysListener.bind(
+      this
+    );
+    this.removeKeyboardHotKeysListener = this.removeKeyboardHotKeysListener.bind(
+      this
+    );
+    this.handleKeyboardHotKeys = this.handleKeyboardHotKeys.bind(this);
+  }
+
+  isToggleHotKey(event) {
+    return [' '].includes(event.key);
+  }
+
+  canTogglePlayPause(event) {
+    // List of elements to do not allow toogle play/pause when pressing space or "k"
+    const disallowedIds = [
+      'recent-song-history',
+      'toggle-play-pause',
+      'stream-select',
+      'keyboard-controls'
+    ];
+    const [focusedElement] = event.path;
+    return !disallowedIds.includes(focusedElement.id);
+  }
+
+  handleKeyboardHotKeys(event) {
+    const keyMap = new Map();
+    keyMap.set(' ', this.togglePlay);
+    keyMap.set('k', this.togglePlay);
+    keyMap.set('ArrowUp', this.increaseVolume);
+    keyMap.set('ArrowDown', this.decreaseVolume);
+
+    if (this.isToggleHotKey(event) && !this.canTogglePlayPause(event)) return;
```

Comment:
```suggestion
    if (this.isSpacePressed(event) && !this.canTogglePlayPause(event)) return;
```
If the function name change is applicable.